src/driver-steelseries.c

```diff
@@ -65,6 +65,74 @@
 #define STEELSERIES_BUTTON_KBD			0x51
 #define STEELSERIES_BUTTON_CONSUMER		0x61
 
+struct steelseries_point {
+	struct list link;
+
+	struct ratbag_color color;
+	uint8_t pos;
+};
+
+struct steelseries_cycle {
+	uint8_t led_id; 			/* led id */
+	uint16_t duration;			/* cycle duration */
+	bool repeat;				/* if the cycle restarts automatically */
+	uint8_t trigger_buttons;	/* trigger button combination */
+	struct list points;			/* colors in the cycle */
+};
+
+static void
+create_cycle(struct steelseries_cycle *cycle)
+{
+	cycle->led_id = 0x00;
+	cycle->duration = 5000;
+	cycle->repeat = true;
+	cycle->trigger_buttons = 0x00;
+
+	list_init(&cycle->points);
+}
+
+static void
+construct_cycle_buffer(struct steelseries_cycle *cycle, uint8_t *buf)
+{
+	struct steelseries_point *point;
+	uint16_t duration, min_duration;
+	uint8_t point_count = 0;
```

Comment:
just a nitpick, I think `npoints` is more natural here but it really doesn't matter :)